crates/api_common/src/request.rs

```diff
@@ -302,6 +302,10 @@ async fn generate_pictrs_thumbnail(
 ) -> Result<Url, LemmyError> {
   let pictrs_config = context.settings().pictrs_config()?;
 
+  if pictrs_config.image_mode() == PictrsImageMode::None {
+    return Ok(image_url.clone());
+  }
+
   if pictrs_config.image_mode() == PictrsImageMode::ProxyAllImages {
     return Ok(proxy_image_link(image_url.clone(), context).await?.into());
   }
```

Comment:
This may be better as a `match` statement.

```suggestion
    match pictrs_config.image_mode() {
        PictrsImageMode::None => return Ok(image_url.clone()),
        PictrsImageMode::ProxyAllImages => return Ok(proxy_image_link(image_url.clone(), context).await?.into()),
        _ => {}
    };
```

I'm not dead-set on either style though. Use which you prefer.